jujugui/static/gui/src/app/app.js

```diff
@@ -1933,6 +1933,16 @@ YUI.add('juju-gui', function(Y) {
       if (this.get('sandbox')) {
         console.log('switching models is not supported in sandbox');
       }
+      // Optionally update to the new base URL.
+      if (this.get('gisf')) {
+        var state = this.state;
+        var baseUrl = state.get('baseUrl');
+        var unsavedModelUrl = 'sandbox';
+        // Make sure we're not already in an unsaved model.
+        if (baseUrl.indexOf(unsavedModelUrl) < 0) {
```

Comment:
Yeah, sorry, the comment is unclear. Two things:

1. "sandbox" is the special URL used in gisf to denote a disconnected GUI.
2. The point here isn't really to see if the model is unsaved. It's more to make sure that the baseUrl update is idempotent; that is, that we're not adding multiple instances of "sandbox" to the URL.